classes/Language.php

```diff
@@ -168,7 +172,14 @@ public function moveToIso($newIso)
             }
         }
 
-        foreach (Theme::getThemes() as $theme) {
+        $theme_manager = new ThemeManager(
```

Comment:
Haven't finished reading the whole thing yet, but if you need to `new` the `ThemeManager` often, maybe a factory (that depends on a `Context`) would be appropriate.